enumerables.rb

```diff
@@ -0,0 +1,116 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    i = 0
+    while i < size
+      yield(self[i])
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    i = 0
+    while i < size
+      yield(self[i], i)
+      i += 1
+    end
+    self
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    i = 0
+    result = []
+    while i < length
+      answer = yield(self[i])
+      result << self[i] unless answer == false
+      i += 1
+    end
+    result
+  end
+
+  def my_all?(pattern = nil)
+    if block_given?
+      my_each { |item| return unless yield item }
+    elsif pattern
+      my_each { |item| result & pattern == item || pattern == item.class }
+    else
+      my_each { |item| result & item }
+    end
+    true
+  end
+
+  def my_any?(pattern = nil)
+    if block_given? && pattern.nil?
+      my_each { |item| return true if item }
+    elsif pattern.is_a? Regexp
+      my_each { |item| return true if item =~ pattern }
+    elsif pattern.is_a? Class
+      my_each { |item| return true if item.is_a? pattern }
+    elsif pattern
+      my_each { |item| return true if item == pattern }
+    elsif !pattern && !block_given?
+      my_each { |item| return true if item }
+    end
+    false
+  end
```

Comment:
my_proc = proc { |num| num > 100 }
[1, 2, 3].my_any?(&my_proc) returns `true` instead of `false`; please fix it too.